virtcontainers/qemu.go

```diff
@@ -1012,6 +1010,12 @@ func (q *qemu) hotplugAddBlockDevice(drive *config.BlockDrive, op operation, dev
 		return err
 	}
 
+	defer func() {
+		if err != nil {
+			q.qmpMonitorCh.qmp.ExecuteBlockdevDel(q.qmpMonitorCh.ctx, drive.ID)
+		}
+	}()
+
 	if q.config.BlockDeviceDriver == config.VirtioBlock {
 		driver := "virtio-blk-pci"
 		addr, bridge, err := q.addDeviceToBridge(drive.ID)
```

Comment:
Good catch!  I think `removeDeviceFromBridge` should be called if `ExecutePCIDeviceAdd` fails, right?

Reply:
You are right! `ExecutePCIDeviceAdd` fails need to rollback too.